platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/com/swirlds/platform/test/gui/TestGuiSource.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.test.gui;
+
+import com.swirlds.common.constructable.ConstructableRegistry;
+import com.swirlds.common.constructable.ConstructableRegistryException;
+import com.swirlds.gui.hashgraph.HashgraphGuiSource;
+import com.swirlds.platform.consensus.ConsensusSnapshot;
+import com.swirlds.platform.gui.hashgraph.internal.FinalShadowgraphGuiSource;
+import com.swirlds.platform.internal.ConsensusRound;
+import com.swirlds.platform.state.signed.SignedState;
+import com.swirlds.platform.state.signed.SignedStateFileReader;
+import com.swirlds.platform.test.consensus.ConsensusUtils;
+import com.swirlds.platform.test.consensus.TestIntake;
+import com.swirlds.platform.test.fixtures.event.EventUtils;
+import com.swirlds.platform.test.fixtures.event.generator.GraphGenerator;
+import com.swirlds.platform.test.fixtures.event.generator.StandardGraphGenerator;
+import com.swirlds.platform.test.fixtures.event.source.EventSource;
+import com.swirlds.platform.test.fixtures.event.source.StandardEventSource;
+import com.swirlds.test.framework.ResourceLoader;
+import com.swirlds.test.framework.context.TestPlatformContextBuilder;
+import java.awt.FlowLayout;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSpinner;
+import javax.swing.SpinnerNumberModel;
+
+public class TestGuiSource {
+    private final GraphGenerator<?> graphGenerator;
+    private final TestIntake intake;
+    private final HashgraphGuiSource guiSource;
+    private ConsensusSnapshot savedSnapshot;
+
+    public TestGuiSource(final long seed, final int numNodes) {
```

Comment:
nullity markings on all arguments and return types in file.

Reply:
done